### PR TITLE
Implement WebView navigation events

### LINKED ISSUE #3440: Implement Navigating, Navigated events in WebViewHandler
### Description

I wanted to check out the new WebView and it turns out the events for when Navigating and Navigated are not doing anything.
As suggested by @mattleibow WebView was switched from renderer to handler and the events were not ported.

Similar PR: https://github.com/dotnet/maui/pull/3316

### Steps to Reproduce

1. Open started MAUI projet
2. Add WebvView anywhere
3. Hook up to Navigating and Navigated events
4. The events are never fired

### Version with bug

Preview 10 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows, I was *not* able test on other platforms

### Affected platform versions

All versions

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Implement WebView navigation events:
- Navigating
- Navigated

Fixes https://github.com/dotnet/maui/issues/3440

### PR Checklist ###

- [x] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)

#### Does this PR touch anything that might affect accessibility?
- No

### CODE REVIEW

File: src/Core/src/Handlers/WebView/WebViewHandler.Android.cs [hunk 1]
```diff
@@ -56,16 +68,23 @@ public static void MapWebViewSettings(WebViewHandler handler, IWebView webView)
 
 		public static void MapGoBack(WebViewHandler handler, IWebView webView, object? arg)
 		{
+			if (handler.NativeView.CanGoBack())
+				handler.CurrentWebNavigationEvent = WebNavigationEvent.Back;
```
Comment (PureWeen) [inline src/Core/src/Handlers/WebView/WebViewHandler.Android.cs:72]: The event state appears to get stuck with whatever it was set to.

- go to gallery for web view
- navigate forward
- click back
- now click any links on the page

You'll get a navigating event with a CurrentWebNavigationEvent set to `Back`

Maybe this call can just be centralized into the handler?

image

**Review by PureWeen — DISMISSED** [review]

File: src/Core/src/Handlers/WebView/WebViewHandler.Android.cs (hunk 1 above)
Comment (jsuarezruiz) [reply src/Core/src/Handlers/WebView/WebViewHandler.Android.cs:72]: Found the issue, this is the pending piece of code: https://github.com/xamarin/Xamarin.Forms/blob/caab66bcf9614aca0c0805d560a34e176d196e17/Xamarin.Forms.Platform.Android/Renderers/FormsWebViewClient.cs#L66

**Review by rmarinho — DISMISSED** [review]
Seems something is making it crash on Android.. but not much info on the output.

Steps to reproduce:

- Go to WebView page
- Click dotnet
- Click GoBack
- Click GoForward
- Click GitHub logo
Crash

On windows when i click back it navigates again to the current page:

Navigating - Url: https://github.com/dotnet/maui, Event: Back
Navigated - Url: https://github.com/dotnet/maui, Event: Back, Result: Success
Navigating - Url: https://github.com/dotnet, Event: NewPage
Navigated - Url: https://github.com/dotnet, Event: NewPage, Result: Success

**Comment by Redth** [comment]
> Seems something is making it crash on Android.. but not much info on the output.
> 
> Steps to reproduce:
> 
> * Go to WebView page
> * Click dotnet
> * Click GoBack
> * Click GoForward
> * Click GitHub logo
>   Crash
> 
> On windows when i click back it navigates again to the current page:
> 
> Navigating - Url: https://github.com/dotnet/maui, Event: Back Navigated - Url: https://github.com/dotnet/maui, Event: Back, Result: Success Navigating - Url: https://github.com/dotnet, Event: NewPage Navigated - Url: https://github.com/dotnet, Event: NewPage, Result: Success

these are fixed with my changes

**Comment by Redth** [comment]
Approving but created https://github.com/dotnet/maui/issues/4938 to track some minor issues on iOS/MacCatalyst 

/cc @mandel-macaque

**Review by Redth — APPROVED** [review]

**Pull request merged**